The distributed jpeg has troubles to be build with newer ndks, starting from
the introduction of the `unified headers` (ndk > 15). This patch allow us to
build the distributed `external jpeg` in sdl package, got the solution in here:
https://github.com/oNaiPs/droidVncServer/issues/53
--- jpeg/Android.mk.orig	2015-06-21 15:14:54.000000000 +0200
+++ jpeg/Android.mk	2019-01-14 10:57:06.384806168 +0100
@@ -20,7 +20,7 @@
 endif

 # temp fix until we understand why this broke cnn.com
-#ANDROID_JPEG_NO_ASSEMBLER := true
+ANDROID_JPEG_NO_ASSEMBLER := true

 ifeq ($(strip $(ANDROID_JPEG_NO_ASSEMBLER)),true)
 LOCAL_SRC_FILES += jidctint.c jidctfst.c
 --- jpeg/jidctfst.S.orig	2019-01-14 11:00:38.000000000 +0100
+++ jpeg/jidctfst.S	2019-01-14 11:00:56.844803970 +0100
@@ -63,7 +63,7 @@


 jpeg_idct_ifast:
-    PLD     [r2, #0]
+    pld     [r2, #0]
     stmdb   sp!, {r4,r5, r6,r7, r8,r9, r10,r11, r12,lr}
     ldr     r4, [sp, #4*10]
     sub     sp, #local_SIZE
@@ -256,7 +256,7 @@

 HLoopStart:
     // reset pointers
-    PLD     [sp, #off_WORKSPACE]
+    pld     [sp, #off_WORKSPACE]
     add     ip, sp, #off_WORKSPACE
     ldr     r10, local_RANGE_TABLE

@@ -268,7 +268,7 @@
     str      r0, local_OUTPUT_BUF
     add      fp, r2, r1

-    PLD      [ip, #32]
+    pld      [ip, #32]
     ldmia    ip!, {r0-r7}

     cmp      r1, #0
